torch_frame/data/multi_nested_tensor.py

```diff
@@ -78,85 +78,161 @@ def from_tensor_mat(
     def __repr__(self) -> str:
         name = ' '.join([
             f"{self.__class__.__name__}(num_rows={self.num_rows},",
-            f"num_cols={self.num_cols},", f"device={self.values.device})"
+            f"num_cols={self.num_cols},", f"device={self.device})"
         ])
-
         return name
 
     def __getitem__(
         self,
         index: Any,
     ) -> Union['MultiNestedTensor', Tensor]:
+        def col_index(
+                multi_nested_tensor: 'MultiNestedTensor',
+                index: Union[int, Tensor, List, slice]) -> 'MultiNestedTensor':
+            r"""Supports all types of column-level advanced indexing of input
+            :class:`MultiNestedTensor` object."""
+            if isinstance(index, int):
+                return multi_nested_tensor.single_index_select(index, dim=1)
+            elif isinstance(index, Tensor):
```

Comment:
There exists a bunch of non-tested code paths here.

Reply:
Added a temporary test on `NotImplementedError`. Will resolve it soon.